### DISCUSSION TITLE
Unable to start medusa js store front on coolify using docker

### DISCUSSION THREAD

**Discussion by Abz2604** [medusajs/medusa#7173] [Q&A]
Hi,

I am relatively new to medusa, nextjs and docker.

For learning purposes I am deploying a nextjs starter storefront on coolify using docker

Dockerfile for the storefront: https://github.com/Abz2604/medusajs-starter/blob/main/bocoai-storefront/Dockerfile

Dockerfile for the backend: https://github.com/Abz2604/medusajs-starter/blob/main/bocoai/Dockerfile

Docker Compose file: https://github.com/Abz2604/medusajs-starter/blob/main/docker-compose.yaml

The deployment happens successfully but I get a 404 on all links and nothing significant that I can see in the Logs to actually debug

I love medusaJS and would really like to get this running so that I can build something meaningful

Help from this wonderful community would be appreciated

-Abz

**Comment by boringcommerce** [comment]
+1, looking for the same thing. Trying to install medusajs (all 3 services, through coolify).

It would be best if there was a one click install in docker platforms like coolify.

**Reply by Abz2604** [reply]
Everything works for me on local, but when deployed using the docker files i'm not sure where the game is going wrong. Any luck?

**Reply by TheSandSensei** [reply]
+1 (having the same exact problem here...and could not resolve)

**Comment by mrfzd** [comment]
Has anyone figured out anything yet?

I watched a video where they set each app individually and linked them together using Coolify's docker link, despite following that line by line I still haven't fixed the same issue you guys having!

I tried a manual deployment using Docker Compose also but that didn't work either!

**Comment by FelipeMiiller** [comment]


**Reply by mrfzd** [reply]
You could go to your project on Coolify and switch to the `Environment Variables` tab and add them all there! The reason it doesn't reads .env file is because of security concerns!

Please let us know if it worked for you, thanks

**Reply by mrfzd** [reply]
<img width="731" alt="image" src="https://github.com/user-attachments/assets/ac8ab231-bfbf-49f0-b250-1430c7574339">

**Reply by FelipeMiiller** [reply]
I use it this way but I can't create a functional docker compose. Could it be the entrypoint that I can enable? It works locally...

**Comment by itsmitul** [comment]
This guide worked to deploy on coolify:
https://docs.medusajs.com/deployments/server/deploying-on-railway

**Comment by AKRking** [comment]
Any success? Team is very inactive on giving self deployment solutions

**Reply by vivasan-tr** [reply]
because of their cloud service

**Comment by sianami123** [comment]
i need install medusa template on coolify too

**Comment by Tudorhc5** [comment]
Hello!

@AKRking  and @sianami123 

You can follow the deployment guide for docker that medusa already has for the backend. Linked [here](https://docs.medusajs.com/learn/installation/docker)

Now if you are a bit tech savy, just start with the [starter front-end tutorial](https://docs.medusajs.com/resources/nextjs-starter) / repository you can figure out how to make little changes to Dockerfile and docker-compose from the the bacend tutorial.

**Reply by AKRking** [reply]
will this work inside coolify? have u tested ?

**Reply by Tudorhc5** [reply]
Yes, I have it deployed as described above.
